### DISCUSSION TITLE
Setup expired time for different cacheName

### DISCUSSION THREAD

**Discussion by baoke344** [ben-manes/caffeine#580] [Q&A] [answered]
Hello!
I want to setup different expire time for each cache-name i have, for example i have 2 cache-name : "product", "customer"
I want each of them have seperate expire time
Is it possibie using expireAfter(Expiry) ? 
Thanks!

**Comment by ben-manes** [comment, marked as answer]
I assume this is for Spring? If so, they'll be your experts on best practices.

In this tutorial [video](https://youtu.be/ze3qFp5pGeA?t=270), a user was able to have independent configurations per cache name in the properties file. That might depend on the Spring Boot version being used, though, as they recently added a [registerCustomCache](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/cache/caffeine/CaffeineCacheManager.html#registerCustomCache-java.lang.String-com.github.benmanes.caffeine.cache.Cache-) method.

An [alternative](https://github.com/stepio/coffee-boots) integration supports this feature directly.

The `expiryAfter(expiry)` idea probably won't work because this library does not have a concept of cache names. That is left to the integration framework, e.g. Spring, Micronaut, Quarkus, who provide the application framework.

**Reply by baoke344** [reply]
Thanks alot!

**Reply by yellowbigbird83** [reply]
Mai I ask how to implement in dropwizard please ?

**Reply by ben-manes** [reply]
I think you implement it manually as there is no cache management abstraction in dropwizard. They use caffeine internally for things like their CachingAuthenticator. You might ask them.
